backend/models/sequence.js

```diff
@@ -60,23 +146,88 @@ const cleanSequenceList = (toClean) => {
 };
 
 const cleanSequenceFrame = (toClean) => {
-	const key = "dateTime";
+	if (toClean.dateTime && utils.isDate(toClean.dateTime)) {
+		toClean.dateTime = new Date(toClean.dateTime).getTime();
+	}
 
-	if (toClean[key] && utils.isDate(toClean[key])) {
-		toClean[key] = new Date(toClean[key]).getTime();
+	if (toClean.viewpoint) {
+		toClean.viewpoint = cleanViewpoint(undefined, toClean.viewpoint);
 	}
 
 	return toClean;
 };
 
-const getSequenceById = async (account, model, sequenceId, projection) => {
-	return db.findOne(account, sequenceCol(model), { _id: sequenceId}, projection);
-};
+const handleFrames = async (account, model, sequenceId, sequenceFrames) => {
+	const processedFrames = [];
+
+	for (let i = 0; i < sequenceFrames.length; i++) {
+		const frame = sequenceFrames[i];
+		const dateTime = new Date(frame.dateTime);
+		let viewpoint;
+
+		if (frame.viewpoint) {
+			viewpoint = await createViewpoint(
+				account,
+				model,
+				undefined,
+				undefined,
+				sequenceId,
+				frame.viewpoint,
+				false,
+				"sequence"
+			);
+		} else {
+			const view = await View.findByUID(account, model, frame.viewId, {});
+
+			if (view) {
+				viewpoint = view.viewpoint;
+
+				["highlighted_group_id",
+					"hidden_group_id",
+					"shown_group_id"
+				].forEach(async (groupIDName) => {
+					if (viewpoint[groupIDName]) {
+						await updateGroup(
+							account,
+							model,
+							undefined,
+							undefined,
+							undefined,
+							undefined,
+							viewpoint[groupIDName],
+							{ sequence_id: sequenceId }
+						);
+					}
+				});
+
+				if (viewpoint["override_group_ids"]) {
+					viewpoint["override_group_ids"].forEach(async (overrideGroupId) => {
+						await updateGroup(
+							account,
+							model,
+							undefined,
+							undefined,
+							undefined,
+							undefined,
+							overrideGroupId,
+							{ sequence_id: sequenceId }
+						);
+					});
+				}
+			} else {
+				throw responseCodes.INVALID_ARGUMENTS;
+			}
+		}
 
-const sequenceExists = async (account, model, sequenceId) => {
-	if(!(await getSequenceById(account, model, utils.stringToUUID(sequenceId), {_id: 1}))) {
-		throw responseCodes.SEQUENCE_NOT_FOUND;
+		processedFrames.push({
+			dateTime,
+			viewpoint
+		});
 	}
+
+	processedFrames.sort(byDateTime);
```

Comment:
byDateTime is only used here and it's so specific to this cause, I think we should just embed the function in here
```suggestion
	processedFrames.sort((frameA, frameB) => frameA.dateTime - frameB.dateTime);
```